version: "3.9"

services:
  python:
    build: .
    image: my_python_image
    restart: on-failure
    depends_on:
      - db
    working_dir: /app
    volumes:
      - ./code:/app/code
      - ./data:/app/data
  db:
    image: postgres:13
    environment:
      - POSTGRES_USER=abdullahahmadfouad
      - POSTGRES_PASSWORD=password123
      - POSTGRES_DB=green_taxi_8_18_postgres
    volumes:
      - "green_taxi_8_18_postgres:/var/lib/postgresql/data/:rw"
      - "m2_queries:/var/lib/m2_queries:rw"
    ports:
      - "5432:5432"
  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin4_container
    restart: always
    ports:
      - "8888:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: abdullahahmadfouad@gmail.com
      PGADMIN_DEFAULT_PASSWORD: password123
volumes:
  green_taxi_8_18_postgres:
  m2_queries:
